### PR TITLE
Fix mentioned users without name and if not in local device

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Links
Resolves https://getstream.slack.com/archives/C04E70P9HJP/p1704309904112239

### 📝 Summary
- Fix not mentioning users if they are not fetched in the local device
- Fix mentioned users without names not highlighted

### 🛠 Implementation
We were using the whole `ChatUser` object in order to mention a user when creating a message. This is not needed, since we only need the user ids. Besides this, with our current architecture, it means that if the `ChatUser` is not available in the local DB, the user would not be mentioned.

#### Fix not mentioning users if they are not fetched in the local device:
Make sure there are no regressions when mentioning a user.

#### Fix mentioned users without names not highlighted
1. Go to a channel with users without names (ex: 4.47.0, `tomasso-XXX`) 
2. Write "@.." to show the mention users suggestions
3. Tap on `tomasso-XXXX`
4. Send the message
5. The message should highlight the user id

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChat/Database/DTOs/MessageDTO.swift [hunk 1]
```diff
@@ -607,13 +612,7 @@ extension NSManagedObjectContext: MessageDatabaseSession {
             }
         )
 
-        // If a user is able to mention someone,
-        // most probably we have that user already saved in DB.
-        // Ideally, this should be `loadOrCreate` but then
-        // we miss non-optional fields of DTO and fail to save.
-        message.mentionedUsers = Set(
```
Comment (martinmitrevski) [inline Sources/StreamChat/Database/DTOs/MessageDTO.swift:614]: What's not clear to me is: when would the `mentionedUsers` value be set, now that we delete this code? We mention in a comment above that this should still be used for reading mentioned users.

**Review by martinmitrevski — COMMENTED** [review]
Just have one question, otherwise looks good.

File: Sources/StreamChat/Database/DTOs/MessageDTO.swift (hunk 1 above)
Comment (nuno-vieira) [reply Sources/StreamChat/Database/DTOs/MessageDTO.swift:614]: This is set on the `saveMessage()` whenever we fetch channels, messages from the servers

**Review by martinmitrevski — APPROVED** [review]
LGTM ✅

**Review by testableapple — APPROVED** [review]
LGTM 🚢

**Pull request merged**